internal/pkg/levee/testdata/src/example.com/tests/namedreturn/tests.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namedreturn
+
+import (
+	"example.com/core"
+)
+
+func TestNamedReturnValue() (s core.Source) {
+	core.Sink(s) // want "a source has reached a sink"
+	return
+}
+
+func TestNamedReturnValuePointer() (s *core.Source) {
+	core.Sink(s) // we do not want a report here, because no Source value has been created
```

Comment:
I agree that we're not expecting a report under current implementation, but I'd consider this a bug in our reliance on `Alloc` rather than a statement of intent.  It's unclear from this comment and this PR's description whether or not you consider this working as intended.

I'd be much happier with a `// TODO want "a source has reached a sink"`, and optionally an explanation as to why we currently don't.

Reply:
I see. Let me try to clarify, then.

Given just this code:
```go
func TestNamedReturnValuePointer() (s *core.Source) {
	core.Sink(s) // we do not want a report here, because no Source value has been created
}
```
There is nothing wrong here, so we don't want a report. Indeed, `s` is `nil`, so it can't be tainted. For `s` to be non-nil, it will have to be assigned a value, i.e. something obtained via a parameter or created locally, in which case we will catch it through those other means. The same logic applies in the following example:
```go
func Test() {
	var s *core.Source
	core.Sink(s) // we don't want a report here either
}
```
We don't want a report here either, because `s` is `nil`. It has never held and will not hold a reference to an actual `core.Source` value.

Does that make sense?

Reply:
I'm sorry.  I really have to stop doing reviews first thin in the morning, before I'm properly caffeinated.  I read through the nilary arguments and was reading `Foo(s *core.Source)`.  I was surprised that you seemed unconcerned that an argument passed by referenced seemed to not be issuing a report.

Now that I'm reading what you actually wrote, yes, I see that you're saying we report on a `var s *Source = nil`.

I think this case might be one that I'm okay with the false positive, at least for now.  So similar to the comment on your other test PR: I'd open an issue about reporting on explicitly nil arguments of source reference type, and add a `// TODO We don't actually want` before the `// want "foo"` that would suppress this test.

Related, would this also report on, say, an empty or `nil` instance of `[]core.Source` or `map[string]*core.Source`?

Reply:
> Now that I'm reading what you actually wrote, yes, I see that you're saying we report on a var s *Source = nil.

We actually *don't* produce a report there, which I think is correct behavior.

Same for `[]core.Source` and `map[string]*core.Source`.

Reply:
My phrasing may have led you onto the wrong path. When I said "we do not want a report here", I didn't mean "we are currently producing a report here, but we shouldn't be".

Reply:
I've now become absolutely and entirely confused...

If we are producing a report here, but shouldn't be, why is Jenkins testing passing?  There's not `// want` there to satisfy the report.

Reply:
To close a long thread of me confusing myself, I'd prefer this comment not use the word `want` due to its strong association elsewhere in our testing.  `// we do not report here, because ...`

Reply:
My bad, forgot to make the update after we discussed it. Thanks for the reminder! Done.